contract/src/lib.rs

```diff
@@ -290,12 +293,38 @@ impl MpcContract {
     }
 
     #[allow(unused_variables)]
-    pub fn sign(&mut self, payload: [u8; 32], path: String) -> [u8; 32] {
-        near_sdk::env::random_seed_array()
+    pub fn sign(&mut self, payload: [u8; 32], path: String) -> Promise {
+        self.pending_requests.insert(&payload, &None);
```

Comment:
This allows two different accounts to submit the same payload simultaneously, but only one of the `sign` calls will return successfully, and possibly with the wrong signature.

`pending_requests` should either be keyed with the payload and the predecessor combined, or `sign` should reject duplicate, simultaneous payloads.